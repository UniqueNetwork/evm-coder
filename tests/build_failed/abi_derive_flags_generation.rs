use evm_coder_procedural::AbiCoderFlags;
use bondrewd::Bitfields;

#[derive(AbiCoderFlags, Bitfields)]
struct EmptyStruct {}


#[derive(AbiCoderFlags, Bitfields)]
struct OneField {
	#[bondrewd(bits = "0..1")]
	a: bool,
}

#[derive(AbiCoderFlags, Bitfields)]
struct ReserveField {
	#[bondrewd(reserve, bits = "0..1")]
	a: bool,
}

#[derive(AbiCoderFlags, Bitfields)]
struct MultipleFields {
	#[bondrewd(bits = "0..1")]
	a: bool,
	#[bondrewd(bits = "1..2")]
	b: bool,
	#[bondrewd(bits = "2..8")]
	c: u8,
}

fn main() {
	assert!(false);
}
