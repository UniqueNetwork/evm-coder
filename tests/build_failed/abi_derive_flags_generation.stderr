error: Empty structs not supported
 --> tests/build_failed/abi_derive_flags_generation.rs:5:8
  |
5 | struct EmptyStruct {}
  |        ^^^^^^^^^^^
